src/qml/controls/BlockClock.qml

```diff
@@ -0,0 +1,175 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtGraphicalEffects 1.0
+
+import "../components"
+
+Item {
+    id: root
+
+    Layout.alignment: Qt.AlignCenter
+    width: size
+    height: size
+
+    property real ringProgress: 0
+    property string header
+    property string subText
+    property bool synced
+    property bool pause
+    property bool conns
+
+    property int headerSize: 32
+
+    property int size: 200
+    property real arcBegin: 0
+    property real arcEnd: ringProgress * 360
+    property real lineWidth: 4
+    property string colorCircle: "#f1d54a"
+    property string colorBackground: Theme.color.neutral2
+
+    property variant blockList: []
+    property variant colorList: ["#EC5445", "#ED6E46", "#EE8847", "#EFA148", "#F0BB49", "#F1D54A"]
+
+    property alias beginAnimation: animationArcBegin.enabled
+    property alias endAnimation: animationArcEnd.enabled
+
+    property int animationDuration: 250
+
+    onArcBeginChanged: canvas.requestPaint()
+    onArcEndChanged: canvas.requestPaint()
+    onSyncedChanged: canvas.requestPaint()
+    onBlockListChanged: canvas.requestPaint()
+
+    Behavior on arcBegin {
+       id: animationArcBegin
+       enabled: true
+       NumberAnimation {
+           duration: root.animationDuration
+           easing.type: Easing.InOutCubic
+       }
+    }
+
+    Behavior on arcEnd {
+       id: animationArcEnd
+       enabled: true
+       NumberAnimation {
+            easing.type: Easing.Bezier
+            easing.bezierCurve: [0.5, 0.0, 0.2, 1, 1, 1]
+            duration: root.animationDuration
+        }
+    }
+
+    Canvas {
+        id: canvas
+        anchors.fill: parent
+        rotation: -90
+
+        onPaint: {
+            var ctx = getContext("2d")
+            var x = width / 2
+            var y = height / 2
+
+            ctx.reset()
+
+            // Paint background
+            ctx.beginPath();
+            ctx.arc(x, y, (width / 2) - parent.lineWidth / 2, 0, Math.PI * 2, false)
+            ctx.lineWidth = root.lineWidth
+            ctx.strokeStyle = root.colorBackground
+            ctx.stroke()
+
+            if (!synced) {
+                var start = Math.PI * (parent.arcBegin / 180)
+                var end = Math.PI * (parent.arcEnd / 180)
+                // Paint foreground arc
+                ctx.beginPath();
+                ctx.arc(x, y, (width / 2) - parent.lineWidth / 2, start, end, false)
+                ctx.lineWidth = root.lineWidth
+                ctx.strokeStyle = root.colorCircle
+                ctx.stroke()
+            }
+
+            else {
+                var del = 0.0025
+                // Paint Block time points
+                for (var i=1; i<parent.blockList.length - 1; i++) {
```

Comment:
The following error appears when going into the paused state.

```
2022-12-17T03:50:17Z GUI: qrc:/qml/controls/BlockClock.qml:96: TypeError: Value is undefined and could not be converted to an object
```

Reply:
Also a small style thing, most prefer to have spaces before and after operators so its easier to read.